### DISCUSSION TITLE
Multiple arguments when first is an array

### DISCUSSION THREAD

**Discussion by akordowski** [spectreconsole/spectre.console#1287] [General]
I am trying to use following arguments:

```
[CommandArgument(0, "<SOURCES>")]
public string[] Sources { get; set; } = default!;

[CommandArgument(1, "<OUTPUT>")]
public string Output { get; set; } = default!;
```

Calling the app `app.exe c:\source1 c:\source2 c:\output` returns following exception:

```
Spectre.Console.Cli.CommandConfigurationException
Message=The command '__default_command' specifies an argument vector that is not the last argument.
```

Is it even possible to have multiple parameters when the first is an array? Whey yes, how can I fix it?
Any help is appreciated.

**Comment by nils-a** [comment]
Only one argument can be an array (named "argument vector") and it needs to be the last argument.

This is currently not documented, but the need to document it is tracked at https://github.com/spectreconsole/spectre.console/issues/644

**Reply by akordowski** [reply]
Thank you for the info!
